extensions/ql-vscode/src/model-editor/model-editor-view.ts

```diff
@@ -441,12 +451,19 @@ export class ModelEditorView extends AbstractWebview<
             ...update,
             message: `Loading models: ${update.message}`,
           }),
-        token: cancellationTokenSource.token,
+        token,
       });
       if (!queryResult) {
         return;
       }
 
+      if (token.isCancellationRequested) {
+        throw new UserCancellationException(
+          "Model editor: Load methods cancelled.",
+          true,
+        );
+      }
+
       this.modelingStore.setMethods(this.databaseItem, queryResult);
     } catch (err) {
       void showAndLogExceptionWithTelemetry(
```

Comment:
I agree that this is where the log message is coming from, so we have the power to silence it. Annoyingly, the error isn't actually a `UserCancellationException` :shakes_fist:. As far as I can tell the error is coming from a rpc library and not our code. We may need to add a custom check for this type of error, like if the message matches `The request \(.*\) has been cancelled`.

Adding in `this.panel?.dispose()` is also good because otherwise the panel lingers open in the "loading" state and you need to close it manually.

So maybe something like
```typescript
if (
  getErrorMessage(err).match(/The request \(.*\) has been cancelled/i)
) {
  this.panel?.dispose();
  return;
}
```

Reply:
> Annoyingly, the error isn't actually a `UserCancellationException`

Is it not? 😅 That `if` snippet worked when I tested locally (though maybe I didn't test all cases 🤔)

Reply:
The if works, but it could be imaginable that something went wrong internally when running the query and an error message with the same text is produced, but we now swallow it.

Reply:
> That `if` snippet worked when I tested locally

It didn't work for me 🤔 . I also tried adding a breakpoint there so I could poke at the error manually. I'm very happy to try again if you disagree.

> it could be imaginable that something went wrong internally when running the query and an error message with the same text is produced, but we now swallow it.

Maybe but I'd expect the message wouldn't say "cancelled" then. How about we still log it but not show the popup? Keeping telemetry could be nice so we know how many people are cancelling this. We could do something like https://github.com/github/vscode-codeql/blob/d0e81124acd4a08f2168298c381023e8de3dbb56/extensions/ql-vscode/src/databases/github-databases/github-databases-module.ts#L66-L71

Reply:
> The if works, but it could be imaginable that something went wrong internally when running the query and an error message with the same text is produced, but we now swallow it.

Sorry, I meant that my original suggestion of `if (err instanceof UserCancellationException)` worked fine for me! I was wondering what Robert meant by saying the error wasn't a `UserCancellationException` 🤔

Reply:
Something which might be less hacky is the following, which checks the type of the other error and that it is a cancellation error:

```ts
import { ResponseError } from "vscode-jsonrpc";
import { LSPErrorCodes } from "vscode-languageclient";

// ...

      if (
        err instanceof ResponseError &&
        err.code === LSPErrorCodes.RequestCancelled
      ) {
        this.panel?.dispose();
        return;
      }
```